txrelayer/txrelayer.go

```diff
@@ -278,15 +289,71 @@ func (t *TxRelayerImpl) waitForReceipt(hash ethgo.Hash) (*ethgo.Receipt, error)
 }
 
 // ConvertTxnToCallMsg converts txn instance to call message
-func ConvertTxnToCallMsg(txn *ethgo.Transaction) *ethgo.CallMsg {
+func ConvertTxnToCallMsg(txn *types.Transaction) *ethgo.CallMsg {
+	gasPrice := uint64(0)
+	if txn.GasPrice() != nil {
+		gasPrice = txn.GasPrice().Uint64()
+	}
+
 	return &ethgo.CallMsg{
-		From:     txn.From,
-		To:       txn.To,
-		Data:     txn.Input,
-		GasPrice: txn.GasPrice,
-		Value:    txn.Value,
-		Gas:      new(big.Int).SetUint64(txn.Gas),
+		From:     ethgo.Address(txn.From()),
+		To:       (*ethgo.Address)(txn.To()),
+		Data:     txn.Input(),
+		GasPrice: gasPrice,
+		Value:    txn.Value(),
+		Gas:      new(big.Int).SetUint64(txn.Gas()),
+	}
+}
+
+// convertTxn converts transaction from types.Transaction to ethgo.Transaction
+func convertTxn(tx *types.Transaction) *ethgo.Transaction {
+	accessList := make(ethgo.AccessList, 0, len(tx.AccessList()))
+
+	for _, e := range tx.AccessList() {
```

Comment:
Make this inline function and call it only when `tx.Type()` is dynamic and access list.